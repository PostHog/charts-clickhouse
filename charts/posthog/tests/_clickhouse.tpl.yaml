# Tests charts/posthog/templates/_clickhouse.tpl
suite: _clickhouse.tpl template
templates:
  - templates/test-templates/clickhouse-env.yaml

tests:
  - it: should render with defaults
    set:
      testTemplates.clickhouseEnvTest: true
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot:
          path: data

  - it: should render with custom values
    set:
      testTemplates.clickhouseEnvTest: true
      clickhouse.cluster: customCluster
      clickhouse.database: customDatabase
      clickhouse.user: customUser
      clickhouse.password: customPassword
      clickhouse.secure: true
      clickhouse.verify: true
    asserts:
      - matchSnapshot:
          path: data

  - it: should fail to render with only clickhouse disabled
    set:
      testTemplates.clickhouseEnvTest: true
      clickhouse.enabled: false
    asserts:
      - failedTemplate:
          errorMessage: externalClickhouse.host is required if not clickhouse.enabled

  - it: should render with external clickhouse
    set:
      testTemplates.clickhouseEnvTest: true
      clickhouse.enabled: false
      externalClickhouse.host: foo.bar.net
      externalClickhouse.user: adminuser
      externalClickhouse.password: hello
    asserts:
      - matchSnapshot:
          path: data

  - it: should render with external clickhouse with more custom settings
    set:
      testTemplates.clickhouseEnvTest: true
      clickhouse.enabled: false
      externalClickhouse.host: foo.bar.net
      externalClickhouse.user: adminuser
      externalClickhouse.password: hello
      externalClickhouse.cluster: customCluster
      externalClickhouse.database: customDatabase
      externalClickhouse.secure: true
      externalClickhouse.verify: true
    asserts:
      - matchSnapshot:
          path: data

  - it: should render with external clickhouse with secrets
    set:
      testTemplates.clickhouseEnvTest: true
      clickhouse.enabled: false
      externalClickhouse.host: foo.bar.net
      externalClickhouse.user: adminuser
      externalClickhouse.existingSecret: someExistingSecret
      externalClickhouse.existingSecretPasswordKey: clickhousePassword
    asserts:
      - matchSnapshot:
          path: data

  - it: should fail to render with external clickhouse but no auth
    set:
      testTemplates.clickhouseEnvTest: true
      clickhouse.enabled: false
      externalClickhouse.host: foo.bar.net
      externalClickhouse.user: adminuser
    asserts:
      - failedTemplate:
          errorMessage: externalClickhouse.password or externalClickhouse.existingSecret is required if using external clickhouse



