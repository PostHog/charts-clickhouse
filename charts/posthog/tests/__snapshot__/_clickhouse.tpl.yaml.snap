should render with custom values:
  1: |
    - name: CLICKHOUSE_HOST
      value: clickhouse-RELEASE-NAME
    - name: CLICKHOUSE_CLUSTER
      value: customCluster
    - name: CLICKHOUSE_DATABASE
      value: customDatabase
    - name: CLICKHOUSE_USER
      value: customUser
    - name: CLICKHOUSE_PASSWORD
      value: customPassword
    - name: CLICKHOUSE_SECURE
      value: "true"
    - name: CLICKHOUSE_VERIFY
      value: "true"
should render with defaults:
  1: |
    - name: CLICKHOUSE_HOST
      value: clickhouse-RELEASE-NAME
    - name: CLICKHOUSE_CLUSTER
      value: posthog
    - name: CLICKHOUSE_DATABASE
      value: posthog
    - name: CLICKHOUSE_USER
      value: admin
    - name: CLICKHOUSE_PASSWORD
      value: a1f31e03-c88e-4ca6-a2df-ad49183d15d9
    - name: CLICKHOUSE_SECURE
      value: "false"
    - name: CLICKHOUSE_VERIFY
      value: "false"
should render with external clickhouse:
  1: |
    - name: CLICKHOUSE_HOST
      value: foo.bar.net
    - name: CLICKHOUSE_PORT
      value: "9000"
    - name: CLICKHOUSE_CLUSTER
      value: somecluster
    - name: CLICKHOUSE_DATABASE
      value: posthog
    - name: CLICKHOUSE_USER
      value: adminuser
    - name: CLICKHOUSE_PASSWORD
      value: hello
    - name: CLICKHOUSE_SECURE
      value: "false"
    - name: CLICKHOUSE_VERIFY
      value: "false"
should render with external clickhouse with more custom settings:
  1: |
    - name: CLICKHOUSE_HOST
      value: foo.bar.net
    - name: CLICKHOUSE_PORT
      value: "9000"
    - name: CLICKHOUSE_CLUSTER
      value: customCluster
    - name: CLICKHOUSE_DATABASE
      value: customDatabase
    - name: CLICKHOUSE_USER
      value: adminuser
    - name: CLICKHOUSE_PASSWORD
      value: hello
    - name: CLICKHOUSE_SECURE
      value: "true"
    - name: CLICKHOUSE_VERIFY
      value: "true"
should render with external clickhouse with secrets:
  1: |
    - name: CLICKHOUSE_HOST
      value: foo.bar.net
    - name: CLICKHOUSE_PORT
      value: "9000"
    - name: CLICKHOUSE_CLUSTER
      value: somecluster
    - name: CLICKHOUSE_DATABASE
      value: posthog
    - name: CLICKHOUSE_USER
      value: adminuser
    - name: CLICKHOUSE_PASSWORD
      valueFrom:
        secretKeyRef:
          key: clickhousePassword
          name: someExistingSecret
    - name: CLICKHOUSE_SECURE
      value: "false"
    - name: CLICKHOUSE_VERIFY
      value: "false"
